### ISSUE TITLE
Export process corrupts .paa image files

### ISSUE THREAD

**Issue by OfficialStamper** [Krzmbrzl/SQDev#181]
When exporting project to arma folders, any .paa image files copied get corrupted and arma will not load them giving the following errors in .rpt file:

`10:55:50 Bad DXT2-5 mipmap size`
`10:55:50 Size out of range`

**Comment by Krzmbrzl** [comment]
Maybe this has something to do with #146. Could you check whether the `.paa` files are valid after they have been imported into the workspace (e.g. by manually copying it into a mission folder from there and try to load it in Arma)?

**Comment by OfficialStamper** [comment]
I am trying to @preserve folders containing any image files without success.  Can what is the correct syntax?

e.g Folder structure:
`\resources\img`

`@preserve "resource"`
or
`@preserve "resource\"`
or
`@preserve "resource\*"`

Help!  ;)

**Comment by OfficialStamper** [comment]
> Maybe this has something to do with #146. Could you check whether the `.paa` files are valid after they have been imported into the workspace (e.g. by manually copying it into a mission folder from there and try to load it in Arma)?

Yes, the files within the eclipse project are valid.  Using Arma 3 Tools | texview2 correctly displays the image.  However, any .paa exported via SQDev cannot be viewed using the same method.

After export, my only option is to completely shut down arma (I think it caches the .paa somewhere), manually replace the .paa files outside of eclipse, restart Arma.

n.b. the links in #146 are broken.

**Comment by Krzmbrzl** [comment]
If you want to use regular expressions for the `@preserve` attribute, you have to prefix the expression:
https://github.com/Krzmbrzl/SQDev/blob/1261f3161beefbb612615c394329ef6e98fb34eb/plugin/Raven.SQDev.Util/src/raven/sqdev/sqdevFile/ESQDevFileAnnotation.java#L237-L244
The respective prefix being `<RegEx>`
https://github.com/Krzmbrzl/SQDev/blob/1261f3161beefbb612615c394329ef6e98fb34eb/plugin/Raven.SQDev.Util/src/raven/sqdev/sqdevFile/ESQDevFileAnnotation.java#L377

If not the name of the folder to be preserved should suffice. Note however that this only prevents the folder from being deleted but not from being overwritten. In order to achieve both, you have to preserve the folder in the target-dir (as you already did) and ignore the folder in the workspace when exporting the project via the `@ignore` attribute. That should do the trick.

> Yes, the files within the eclipse project are valid. Using Arma 3 Tools | texview2 correctly displays the image. However, any .paa exported via SQDev cannot be viewed using the same method.

Hm alright... That's very weird though. I am using Java's `Files.copy` method to copy files around.
https://github.com/Krzmbrzl/SQDev/blob/1261f3161beefbb612615c394329ef6e98fb34eb/plugin/Raven.SQDev.Misc/src/raven/sqdev/misc/FileSystemUtil.java#L355
and I assumed that this shouldn't change the copied files.
And since you are trying RegExes in the attributes, I assume you are on the dev-build of the plugin?
If not then I think this is indeed a duplicate of the referenced issue. In that case I'd advise you to get the latest dev-build available through the SQDev-Discord in `#testing` (its pinned to the channel)

**Comment by OfficialStamper** [comment]
@Krzmbrzl I'm using 0.7.7 from the Releases tab, I suspect that is not your dev branch.

I do not see a dev branch, only Master, gh-pages and plugin.

If there is another version other than 0.7.7 that I could use, please advice.  Happy to give feedback.

**Comment by Krzmbrzl** [comment]
Ah yeah that could be the problem then. As I wrote above the dev-version can only be downloaded via Discord from the `#testing` channel

**Comment by OfficialStamper** [comment]
[note to self, always read the small print]

Yup, got your Dev from Discord, installing now.  Will let you know, thanks for the quick and great response.
